rfcs/framework-0072-pluggable-transfers.md

```diff
@@ -0,0 +1,507 @@
+```python
+def get_form_config(**kwargs):
+    """Get the react form configuration."""
+   return dict(
+    #    ...
++       default_transfer_type=current_transfer_registry.default_transfer_type,
++       transfer_types=list(current_transfer_registry.get_transfer_types()),
+        **kwargs,
+    )
+```
+
+#### Deposit form app
+
+In `DepositFormApp` we drop the concept of always defaulting to `RDMDepositFiles*`.
+We default to using [services registered by Files UI entrypoint from above](https://github.com/oarepo/invenio-rdm-records/commit/a0adca214b279bff0f9ec546709b6fbb5aa5f39f) instead.
+
+Wherever we rendered the original `FileUploader` React component, we now
+[substitute it with the one from Files UI entrypoint](https://github.com/oarepo/invenio-app-rdm/commit/ca6959c2c3331825c73ced17a95086ccaf917e3e). There's still a possibility to override it, or its sub-component parts using
+`react-overriable`.
+
+#### Invenio-RDM-Records module exports
+
+To allow other Files UI implementation build on available Files Redux store actions, or possibly reusable
+file components (e.g. `FileListTable`), they now need to be exported by the `@js/invenio_rdm_records` module.
+
+### Custom Files UI with Uppy
+
+Here we describe integration of [Uppy.io](https://uppy.io/) as a custom Files UI implementation addressing
+use cases regarding reliable multi-part uploads.
+
+<!-- TODO: move under inveniosoftware org? -->
+Uppy integration is being developed as the [`invenio-files-uppy-js`](https://github.com/oarepo/invenio-files-uppy-js/tree/main) NPM package. It provides its [own implementation of `DepositFiles*` services](https://github.com/oarepo/invenio-files-uppy-js/blob/main/src/api/index.js), and the `UppyFileUploader` field component, based off [Uppy Dashboard](https://uppy.io/docs/dashboard/) UI. For the multipart uploading mechanism, it uses
+[AWS S3 Multipart](https://www.npmjs.com/package/@uppy/aws-s3-multipart) package, adapted to match REST APIs designed by this RFC by the [InvenioMultipartUploader](https://github.com/oarepo/invenio-files-uppy-js/blob/main/src/components/UppyUploader/InvenioMultipartUploader.js) Uppy plugin.
+
+Drag & drop file input with auto-thumbnails, upload progress tracking, upload abortion & retry and status bar is provided out of the box. Files Redux store actions is still being used to manage record files state.
+
+More features & integrations like the remote file sources, image or file metadata editor, could be potentially added to the Dashboard UI later.
+
+#### Usage
+
+Uppy integration could be enabled by registering it from a [Webpack Files UI entrypoint](#webpack-files-ui-entrypoint) and then setting it to be used from the site config:
+```python
+FILES_UI_ENTRYPOINT = "invenio-app-rdm-files-uppy"
+```
+
+#### UI workflow
+
+Default dashboard state:
+![Screenshot 2025-02-13 at 8 57 20](https://github.com/user-attachments/assets/85319625-7510-4482-b314-64eac835fc73)
```

Comment:
These look great, especially the auto-thumbnails :)

I feel there's a bit of information and functionality duplication with having both the tiles and table below. 

I see the Uppy Dashboard components have decent customization, so I'm not that worried about being able to sync/combine the two for a good UX.